minhashcuda.cc

```diff
@@ -302,35 +302,32 @@ static std::vector<uint32_t> calc_best_split(
   // We evaluate each variant by the cumulative cost function.
   // Every call to mhcuda_calc() can grow the buffers a little; the cost function
   // optimizes for the number of reallocations first and the imbalance second.
-
+  if (devs.size() == 1) {
+    std::vector<uint32_t> v = {length};
+    return v;
+  }
   uint32_t ideal_split = rows[length] / devs.size();
-  std::vector<std::vector<uint32_t>> variants;
-  for (size_t devi = 0; devi < devs.size(); devi++) {
-    uint32_t row = std::upper_bound(
-        rows, rows + length + 1, ideal_split * (devi + 1)) - rows;
+  std::vector<std::vector<uint32_t>> variants = {{}};
```

Comment:
Are you sure you need the first empty variant?

Reply:
yes, or we don't go into the nested for loop that follow